src/core/Tokenizer.js

```diff
@@ -49,6 +50,10 @@ export default class Tokenizer {
         return new RegExp(`^(${reservedWordsPattern})\\b`, "i");
     }
 
+    createWordRegex(specialChars = []) {
+        return new RegExp(`^([\\w|${specialChars.join("|")}]+)`);
```

Comment:
This will result in regex like: `^([\w|@|#]+)`

As the `|` characters are part of character class `[ ... ]` this will match stuff like `foo|bar` as a single word, as `|` has no special meaning inside square brackets.